File: pinyougou-sellergoods-interface/src/main/java/com/lsf/pinyougou/sellergoods/service/SellerService.java
package com.lsf.pinyougou.sellergoods.service;

import java.util.List;

import com.lsf.pinyougou.pojo.TbSeller;
import vo.PageResult;


/**
 * 服务层接口
 */
public interface SellerService {

    /**
     * 返回全部列表
     *
     * @return
     */
    List<TbSeller> findAll();


    /**
     * 无条件分查询
     *
     * @return
     */
    PageResult findPage(int pageNum, int pageSize);


    /**
     * 多条件分页查询
     *
     * @param pageNum  当前页
     * @param pageSize 每页记录数
     * @return
     */
    PageResult findPageLimit(TbSeller seller, int pageNum, int pageSize);


    /**
     * 添加
     */
    void add(TbSeller seller);


    /**
     * 修改商家的审核状态
     */
    void update(TbSeller seller);


    /**
     * 根据 ID 查询某个未审核商家的全部信息
     *
     * @param id
     * @return
     */
    TbSeller findOne(String id);


    /**
     * 批量删除
     *
     * @param ids
     */
    void batchDelete(Long[] ids);

}

File: pinyougou-manager-web/src/main/webapp/js/service/sellerService.js
// 商家管理服务层
pyg.service('sellerService', function ($http) {

    // 查询所有数据
    this.findAll = function () {
        return $http.get('../seller/findAll.do');
    };

    // 无条件分页查询
    this.findPage = function (page, size) {
        return $http.get('../seller/findPage.do?page=' + page + '&size=' + size);
    };

    // 根据 ID 查询某个未审核商家的全部信息
    this.findOne = function (id) {
        return $http.get('../seller/findOne.do?id=' + id);
    };

    // 商家入驻申请
    this.add = function (entity) {
        return $http.post('seller/add.do', entity);
    };

    // 修改商家的审核状态
    this.update = function (entity) {
        return $http.post('../seller/update.do', entity);
    };

    // 批量删除
    this.batchDelete = function (selectedList) {
        return $http.get('../seller/batchDelete.do?ids=' + selectedList);
    };

    // 多条件分页查询
    this.findPageLimit = function (page, size, searchEntity) {
        return $http.post('../seller/search.do?page=' + page + "&size=" + size, searchEntity);
    };
});
